utils/webhook.js

```diff
@@ -0,0 +1,76 @@
+const { getResponseJSON } = require("./shared");
+const { processTwilioEvent, processSendGridEvent } = require("./firestore");
+const { SecretManagerServiceClient } = require("@google-cloud/secret-manager");
+const { EventWebhook, EventWebhookHeader } = require("@sendgrid/eventwebhook");
+
+/* This function will process the webhook data from Twilio */
+const handleReceivedTwilioEvent = async (req, res) => {
+    try {
+        await processTwilioEvent(req.body);
+
+        return res.status(200).json({ code: 200 });
+    } catch (e) {
+        console.error("twilioSmsEventWebhook error", e);
+        return res
+            .status(500)
+            .json(getResponseJSON("Internal Server Error!", 500));
+    }
+};
+
+const handleReceivedSendGridEvent = async (req, res) => {
+    try {
+        const client = new SecretManagerServiceClient();
+        const [version] = await client.accessSecretVersion({
+            name: process.env.GCLOUD_SENDGRID_EVENT_WEBHOOKSECRET,
+        });
+        const publicKey = version.payload.data.toString();
+        const eventWebhook = new EventWebhook();
+        const ecPublicKey = eventWebhook.convertPublicKeyToECDSA(publicKey);
```

Comment:
Better move secret setups out, so that these lines (line 22-28) don't need to run each time a request comes in.

Reply:
Sendgrid recommends that we validate each request from them. so i think it's necessary.